### PR TITLE
Prioritization tools

### PR DESCRIPTION
**Pull request by zoeblairfriedman**
This PR covers the following quicksilver line items in the [Prioritization Tools](https://quicksilver.engineering.mercuryanalytics.com/tasks/69) task:
- Add an "approved" checkbox to tasks to indicate which wishlist items are ready to promote.
- Make a reasonable way to promote tasks from the wishlist.

More specifically:
- Adds checkbox to form partial that can only be toggled by non-engineers if it is a wishlist task
- Adds column to relevant boards showing the approved state via a red "X" or green check mark.
- Adjusts permissions so that only engineers can change boards and dates
- Adjusts permissions so that only engineers can create board-less tasks
- Adds relevant specs

### CODE REVIEW

File: spec/features/task_list_spec.rb [hunk 1]
```diff
@@ -5,26 +5,76 @@
...
-    create_list(:task, 3, completed_at: 3.days.ago)
   end
 
-  it "shows all the tasks" do
-    visit tasks_path
+  context "when visiting index of tasks without board" do
+    before do
+      create_list(:task, 2, completed_at: nil)
+      create_list(:task, 3, completed_at: 3.days.ago)
+      create(:task, title: "Boardless Task")
```
Comment (m4thayus) [inline spec/features/task_list_spec.rb:14]: Probably overkill to hardcode the name here to be tested later if you had a let that was the "boardless list" then you can just test `boardless_list.first.name` or similar

**Review by m4thayus — CHANGES_REQUESTED** [review]
We can discuss details in the morning

File: app/views/tasks/index.html.slim [hunk 2]
```diff
@@ -1,6 +1,12 @@
+- @current_action = controller.request.path_parameters[:action].to_sym
+
 h1 List of Tasks
 
 - unless @active_tasks.empty?
+  - if @board
+    h2
+      = @board.name
```
[the remark this answers is not in the document]
Comment (zoeblairfriedman) [reply app/views/tasks/index.html.slim:8]: having trouble with this

File: spec/models/ability_spec.rb [hunk 3]
```diff
@@ -24,13 +24,39 @@
...
+    it { is_expected.to_not be_able_to(:update, build(:task), :approved) }
+    it { is_expected.to_not be_able_to(:create, build(:task, board: wishlist), :started_at) }
+    it { is_expected.to_not be_able_to(:create, build(:task, board: wishlist), :expected_at) }
+    it { is_expected.to_not be_able_to(:create, build(:task, board: wishlist), :completed_at) }
+    it { is_expected.to_not be_able_to(:create, build(:task, board: wishlist), :board_id) }
+    it { is_expected.to_not be_able_to(:update, wishlist_task, :started_at) }
+    it { is_expected.to_not be_able_to(:update, wishlist_task, :expected_at) }
+    it { is_expected.to_not be_able_to(:update, wishlist_task, :completed_at) }
+    it { is_expected.to_not be_able_to(:update, wishlist_task, :board_id) }
   end
```
[the remark this answers is not in the document]
Comment (zoeblairfriedman) [reply spec/models/ability_spec.rb:62]: Does this feel more reasonable now?

File: app/views/tasks/_table.html.slim [hunk 4]
```diff
@@ -9,6 +9,12 @@ table
       - %w[completed_on actual_duration delta].each do |header|
         th
           = header.titleize
+    - if @board && @board.name == "wishlist"
```
Comment (m4thayus) [inline app/views/tasks/_table.html.slim:12]: this is just `@board&.name`

File: app/views/tasks/_table.html.slim [hunk 5]
```diff
@@ -43,3 +49,6 @@ table
                 '  over
               - else
                 '  under
+        - unless show_statistics
+          td
+            input class="approved" type="checkbox" checked=task.approved disabled=!task.approved
```
Comment (m4thayus) [inline app/views/tasks/_table.html.slim:54]: use readonly https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#readonly

File: app/assets/stylesheets/main.scss [hunk 6]
```diff
@@ -12,6 +12,13 @@ table {
     font-weight: bold;
     margin-bottom: 0.75rem;
   }
+  td:has(.approved) {
+    text-align: center;
+    input[type="checkbox"] {
+      pointer-events: none;
+      accent-color: #059033;
+    }
+  }
```
Comment (m4thayus) [inline app/assets/stylesheets/main.scss:21]: Not quite what I was talking about let's work on this together

File: spec/models/ability_spec.rb (hunk 3 above)
[the remark this answers is not in the document]
Comment (m4thayus) [reply spec/models/ability_spec.rb:62]: Much better 😃

File: app/assets/stylesheets/main.scss (hunk 6 above)
Comment (zoeblairfriedman) [reply app/assets/stylesheets/main.scss:21]: shoutout to using ":has" though. Pretty cool.

**Review by m4thayus — APPROVED** [review]

**Pull request merged**